app/Http/Controllers/Eportalcontroller.php

```diff
@@ -85,33 +98,32 @@ public function store(Request $request)
             $file->save();
             if($request->abc != null)
             {
-                return redirect('/show?abc='.$file->parent_directory)->with("success","Successfully Uploaded");
+                return redirect('/files'.'/'. $file->parent_directory)->with("success","Folder Created");
             }
             else{
-                return redirect('/files');
+                return redirect('/files')->with('success','Folder Created');
             }
             
         }
        
        
-    }
-          
-    
-    
-
+    }   
+   
     /**
      * Display the specified resource.
      *
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function show(Request $request)
+    public function show($filesID)
```

Comment:
there is something called type-hinting in PHP.
```suggestion
    public function show(integer $fileId)
```

Reply:
Additionally, Laravel provides route-model binding for resources. Which means you can do something like:
```suggestion
    public function show(Directory $directory)
```

In this case, you won't have to write Directory::find